substra/sdk/backends/local/dal.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2018 Owkin, inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pathlib
+import tempfile
+import typing
+
+from substra.sdk import exceptions, schemas
+from substra.sdk.backends.remote import backend
+from substra.sdk.backends.local import db, models
+
+
+class DataAccess:
+    """Data access layer.
+
+    This is an intermediate layer between the backend and the local/remote data access.
+    """
+
+    def __init__(self, remote_backend: typing.Optional[backend.Remote]):
+        self._db = db.get()
+        self._remote = remote_backend
+        self._tmp_dir = tempfile.TemporaryDirectory(prefix="/tmp/")
+
+    @property
+    def tmp_dir(self):
+        return pathlib.Path(self._tmp_dir.name)
+
+    @staticmethod
+    def _is_local(key: str):
+        """Check if the key corresponds to a local
+        or remote asset.
+        """
+        return key.startswith("local_")
+
+    def _get_asset_name(self, type_):
```

Comment:
Maybe get_asset_content_filename would be more descriptive?

Reply:
Yes, better !